components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -84,6 +89,69 @@ void ZCashWalletService::GetZCashAccountInfo(
   std::move(callback).Run(keyring_service_->GetZCashAccountInfo(account_id));
 }
 
+void ZCashWalletService::MakeAccountShielded(
+    mojom::AccountIdPtr account_id,
+    MakeAccountShieldedCallback callback) {
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (IsZCashShieldedTransactionsEnabled()) {
+    if (!GetAccountShieldBirthday(account_id).is_null()) {
+      std::move(callback).Run("Already shieldable");
+      return;
+    }
+    GetLatestBlockForAccountBirthday(account_id.Clone(), std::move(callback));
+    return;
+  }
+#endif
+  std::move(callback).Run("Not supported");
+}
+
+void ZCashWalletService::StartShieldSync(
+    mojom::AccountIdPtr account_id,
+    mojo::PendingRemote<mojom::ZCashSyncObserver> observer,
+    StartShieldSyncCallback callback) {
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (IsZCashShieldedTransactionsEnabled()) {
+    auto account_birthday = GetAccountShieldBirthday(account_id);
+    if (!account_birthday) {
+      std::move(callback).Run("Account not supported");
+      return;
+    }
+
+    auto it = shield_sync_services_.find(account_id);
+    if (it == shield_sync_services_.end()) {
+      shield_sync_services_[account_id.Clone()] =
```

Comment:
Let's make it fail if account is already being synced.

Reply:
Done